### ISSUE TITLE
[FEATURE] Use `warp.jax_callable` on CPU

### ISSUE THREAD

**Issue by jeertmans** [jeertmans/DiffeRT#476]
### Terms

- [x] Checked the [existing issues](https://github.com/jeertmans/DiffeRT/issues?q=is%3Aissue+label%3Aenhancement) to see if my suggestion has not already been suggested;

### Description

As of #467, we use `jax.lax.platform_dependent` + `jax.pure_callback` as a (temporary) to support both GPUs and CPUs. Once https://github.com/NVIDIA/warp/pull/1446 is merged, we should be able to simplify the logic by only having one implementation for all platforms.
